### ISSUE TITLE
Hanging on Configuring SSH to Master Node

### ISSUE THREAD

**Issue by bmccann** [jtriley/StarCluster#272]
I would like to setup a system of nodes as follows:

1 master node with two EBS volumes attached, one of which is configured to host a mySQL database for the master.

An auto-scaling number of workers with NFS access to the database EBS volume on the master. I would prefer not to give them NFS access to the other volume, but if that is complicated I could mount them both with StarCluster and allow NFS access to both volumes.

Steps I took:
I created an AMI for the worker nodes.
I also created two EBS volumes, both of which I attached to another node with all of the software necessary for the master.
After configuring the database volume to house the mySQL database for the master node (http://aws.amazon.com/articles/1663) and adding the necessary code to the other EBS volume, I used the AWS console to create an AMI for the master. Before clicking create, I removed the two EBS volumes in the Create Image box so that they would not be included in the AMI. 

Upon attempting to start a new cluster from the Master AMI, with any number of nodes (including at least the master, of course) and whether or not I attach the two EBS volumes, startup hangs at 

> > > Waiting for SSH to come up on all nodes...

when it only has one node left. 

I can open another terminal and log into all nodes except the master.

**Comment by jtriley** [comment]
@bmccann Sorry for the delay on this. It's hard to tell what's going on without looking at your AMI(s) but it sounds like it's an issue with your master AMI. Are you able to launch this AMI outside of StarCluster and login to the instance? 

Also, for the master AMI, which AMI did you base this off of? Does this happen every time you try to start the cluster or just that once? If you're able to reproduce this every time please post the output of:

```
$ starcluster listclusters --show-ssh-status
```

and

```
$ starcluster showconsole i-999999
```

You should replace `i-999999` with the master's instance id (see the `listclusters` output)

**Comment by JaeYJung** [comment]
I had a similar problem. I started my cluster with 5 nodes, and "Waiting for SSH" hung on 4/5.

However, "listclusters -s" (on another terminal) showed that all SSH were up, and I could ssh/login all five nodes. 

All were spot instances, compute nodes with ami-765b3e1f (ubuntu 64bit), and master node was with my own ami based on ami-765b3e1f. I have had no problem booting with this ami as a single instance. I am using develop branch of StarCluster.

**Comment by viktor2** [comment]
I had a similar issue.  I also mounted ebs volume and unmounted before creating ami.  When I tried to bring up a cluster based on that ami ebs volume initialization attempted to mount the volume again for some reason.  It failed since the volume had not been mounted at that point.  Using $ starcluster showconsole i-999999 clearly showed that this was the problem.  I was able to resolve it by creating a script to perform the mounts but not actually mounting the instance that I converted to ami.  Every time I bring up instances based on this ami I run the script that mounts mysql directories on ebs volume.  The important part is to do mysql mount after ami is fully initialized and ebs volume is attached.

**Comment by bmccann** [comment]
Sorry for leaving this open so long. I did end up mounting after the image was created as viktor2 described. This worked well for a while, but eventually I switched to using Amazon's RDS instances. Thanks for the help everyone!

**Comment by diego898** [comment]
what was the solution? I am having a similar problem

**Comment by alemagnani** [comment]
I am having the same problem as well

**Comment by jtriley** [comment]
@alex @diego898 Please include more details on your cluster setup. Are you using the latest 0.95.2 release? Also, are you using VPC by chance?

**Comment by diego898** [comment]
I am using the latest release and am not using vpc.

I started with one of the default ubuntu amis starcluster has. I updated, and installed the GSL, and created a new AMI: ami1. I then started from ami1, installed a different python suite (anaconda), removed OpenMPI,leaving me with mpich2. Ran some simple tests with c/mpi and python/mpi4py. Seemed everthing was fine. I then created a new ami from this: ami2. Whenever I try and start a cluster from ami2 I can never connect and ssh never comes up. Ive repeated these steps twice forming ami3 and ami4. Both ami2 and ami4 dont work, while ami1 and ami3 do.
